src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,18 +3,41 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int DEFAULT_SIZE = 10;
+    private final K[] keysArray = (K[]) new Object[DEFAULT_SIZE];
+    private final V[] valuesArray = (V[]) new Object[DEFAULT_SIZE];
+    private int size = 0;
 
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < size; i++) {
+            if ((key == null
+                    && keysArray[i] == null)
+                    || key != null && key.equals(keysArray[i])) {
```

Comment:
```suggestion
            if ((key ==  keysArray[i]) || key != null && key.equals(keysArray[i])) {
```
це можна спростити, враховуючи, що якщо два об'єкти нали, то порівняння по посиланнях нам дасть правильний результат